inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api_impl.cpp

```diff
@@ -470,6 +471,23 @@ std::string InferenceEnginePython::get_version() {
     return version_str;
 }
 
+InferenceEnginePython::IENetwork InferenceEnginePython::read_network(const std::string & path_to_xml,
+                                                                     const std::string & path_to_bin,
+                                                                     const std::vector<std::string> & extensions) {
+    std::vector<InferenceEngine::IExtensionPtr> ext;
+    for (const auto & path : extensions) {
+        ext.push_back(std::make_shared<InferenceEngine::Extension>(path));
+    }
+
+    auto net = InferenceEngine::details::ReadNetwork(path_to_xml, path_to_bin, ext);
+    return InferenceEnginePython::IENetwork(std::make_shared<InferenceEngine::CNNNetwork>(net));
+}
+
+InferenceEnginePython::IENetwork InferenceEnginePython::read_network_without_extensions(const std::string & path_to_xml,
+                                                                                        const std::string & path_to_bin) {
+    auto net = InferenceEngine::details::ReadNetworkWithoutExtensions(path_to_xml, path_to_bin);
+    return InferenceEnginePython::IENetwork(std::make_shared<InferenceEngine::CNNNetwork>(net));
```

Comment:
can we use extension API to read custom nodes to `FrameworkNode`?

Reply:
No just because operation may have any opset and type but extension API requires opset to be set to "experimental" or some other value but not to arbitrary value.

_Discussed this with @ilyachur. He is currently agreed with this approach._

Reply:
Looks strange to discuss in a small group and forcing the decision on others.

Reply:
Maybe can we add hardcoded operation set with one operation (like: {"__openvino_default_op__": {FrontEndNode}}) and if user add extension with this name, it means that readers can read the network with custom ops

Reply:
it's not ideal solution, but explicit implication at least allows to use default IECore and default extensions mechanism. 

Let's try this approach.

Reply:
This is not a new reader API. At least it is not a public API change. We are modifying internal API for Reader to unlock some additional functionality that is going to be publicly available via FE API. As we are not ready to introduce this FE API immediately, @GlebKazantaev has found a minimal solution with creating this tiny hole. Eventually core IR reader code should migrate to ngraph/frontend and MO will use it directly, particularly propose method with FrontEndNode is going to be implemented as a part of FrontEnd::convertPartially method. The public API of IReader is not going to be changed (and not changed in this PR).

What's wrong with adding new functionality to our hidden classes to support our own tools like Model Optimizer? What are real drawbacks? I mean, ideally, sure, Model Optimizer should rely on public API only, but different components are moving with different speed in the development process. The proposed solution with openvino_default_op is more like a hack with this hardcoded name that may affect public flow. In contrast, the idea of the original PR is to modify internal class and access it directly without disturbing any really public APIs.

Reply:
Thanks to @GlebKazantaev, realized that we actually do public API change. I was wrong. Why cannot we call the IR reader code directly without modifying the public API?

Reply:
@ilya-lavrenov, I'm not forcing decisions to others. I just noted Ilya C. opinion after our offline discussion that he currently agree with this solution.

@slyalin I assume that technically it is possible but it also requires IRReader header to be available in python bindings library. And as a result we still have this dependency with internal API.

If I understand right, in Frontend API all frontends implement `convert_partially` method which wraps unsupported ops with FrameworkNode. And from my point of view `convert_partially` simply matches `read_network_without_extensions` which is needs to be implemented by readers. So I don't see any objections why should we search for another solution. I understand @ilya-lavrenov concern about internal API dependency, but this solution is temporary and needed only for 21.4 release. Starting from 22.1 we will have Python Frontend API which will replace this work-around.

Also I  wonder, how the solution with extension supposed to simplify our life? I think it is overkill to build library with extension which is needed only to enable custom operation support. And how should we search for this library inside Model Optimizer? And finally, it still requires IR/ONNX readers to implement this logic.

Reply:
Hi guys,

I think we need to create a meeting and discuss the issue! 😄 

The main problem of this PR is using private API (it is not API changes). When we discussed with Gleb, I got his position and didn't have good ideas how we can avoid it.

Yesterday I proposed to use special extension. It allows to use only public API.
You shouldn't build a separate extension library, you need to have `FrontendNode` only in your code.
If in 22.1 we have Frontend API which doesn't have such issues, I think that the proposal with extension creates a less impact on components.

Reply:
**Offline discussion summary:** special extension will be used to trigger `FrameworkNode` usage inside IRReader. In future releases this temporary solution will be replaced with low-level Frontend python API.

_Attendees: @ilya-lavrenov @echuraev @slyalin_  

Mark this discussion as resolved.